src/load/azext_load/data_plane/utils/utils.py

```diff
@@ -340,12 +344,66 @@ def convert_yaml_to_test(cmd, data):
     if data.get(LoadTestConfigKeys.AUTOSTOP) is not None:
         new_body["autoStopCriteria"] = utils_yaml_config.yaml_parse_autostop_criteria(data=data)
 
-    utils_yaml_config.update_engine_reference_identity(new_body, data)
+    utils_yaml_config.update_reference_identities(new_body, data)
     logger.debug("Converted yaml to test body: %s", new_body)
     return new_body
 # pylint: enable=line-too-long
 
 
+# pylint: disable=line-too-long
+# Disabling this because if conditions are too long
+def parse_app_comps_and_server_metrics(data):
+    app_components_yaml = data.get(LoadTestConfigKeys.APP_COMPONENTS)
+    app_components = {}
+    server_metrics = {}
+    add_defaults_to_app_copmponents = dict()
+    if app_components_yaml is None:
+        return None, None, None
+    if not isinstance(app_components_yaml, list):
+        raise InvalidArgumentValueError("App component name should be of type list")
+    for app_component in app_components_yaml:
+        resource_id = app_component.get(LoadTestConfigKeys.RESOURCEID)
+        if not isinstance(app_component, dict):
+            raise InvalidArgumentValueError("App component name should be of type dictionary")
+        if app_component.get(LoadTestConfigKeys.RESOURCEID) is None:
+            raise InvalidArgumentValueError("App component name is required")
+        if not is_valid_resource_id(app_component.get(LoadTestConfigKeys.RESOURCEID)):
+            raise InvalidArgumentValueError("App component name is not a valid resource id")
+        if add_defaults_to_app_copmponents.get(resource_id.lower()) is None:
+            add_defaults_to_app_copmponents[resource_id.lower()] = app_component.get(LoadTestConfigKeys.SERVER_METRICS_APP_COMPONENTS) is None
+        else:
+            add_defaults_to_app_copmponents[resource_id.lower()] = add_defaults_to_app_copmponents.get(resource_id.lower()) and app_component.get(LoadTestConfigKeys.SERVER_METRICS_APP_COMPONENTS) is None
```

Comment:
Should this be `or` ?

Reply:
no its correct, becasuse multiple app-comps can be given and one of them have the metrics, then we should not default to the default metrics, so its and only.
![image](https://github.com/user-attachments/assets/1aac88b7-0ea3-4688-8656-94d69f90ca8f)
ado for refewrence.